src/classes/RD2_RecurringDonations_TEST.cls

```diff
@@ -332,34 +334,66 @@ private with sharing class RD2_RecurringDonations_TEST {
     */
     @isTest
-    private static void shouldOverrideStartDateWithCurrentDate() {
-        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
-
-        Contact contact = getContact();
-
-        npe03__Recurring_Donation__c rd = TEST_RecurringDonationBuilder.constructEnhancedBuilder()
-            .withContact(contact.Id)
-            .withAmount(50)
-            .withInstallmentPeriod(RD2_Constants.INSTALLMENT_PERIOD_MONTHLY)
-            .withDayOfMonth('10')
-            .withInstallmentFrequency(1)
-            .withStatus(RD2_Constants.STATUS_ACTIVE)
-            .withDateEstablished(Date.newInstance(2019, 6, 5))
-            .withStartDate(Date.newInstance(2019, 6, 5))
+    private static void shouldNotOverrideStartDateWithCurrentDate() {
+        Date startDate = Date.newInstance(2019, 10, 5);
+        npe03__Recurring_Donation__c rd = getRecurringDonationMonthlyBuilder()
+            .withStartDate(startDate)
             .build();
 
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
         Test.startTest();
         insert rd;
-        rd.npe03__Amount__c = 100;
+        rd.npe03__Amount__c = 250;
         rd.InstallmentFrequency__c = 2;
         update rd;
         Test.stopTest();
 
+        List<RecurringDonationSchedule__c> rdSchedules = rdScheduleGateway.getRecords(rd);
+        System.assertEquals(2, rdSchedules.size(), 'Should be 2 schedules.');
+
+        System.assertEquals(startDate, rdSchedules[0].StartDate__c, 'Original start date should be 10/5/2019.');
+        System.assertEquals(startDate.addDays(-1), rdSchedules[0].EndDate__c, 'Original end date should be 10/4/2019.');
+        System.assertEquals(startDate, rdSchedules[1].StartDate__c, 'Updated start date should be 10/5/2019.');
+        System.assertEquals(null, rdSchedules[1].EndDate__c, 'Updated end date should be null.');
+        System.assertEquals(100, rdSchedules[0].InstallmentAmount__c, 'Original amount should equal 100.');
+        System.assertEquals(250, rdSchedules[1].InstallmentAmount__c, 'Updated amount should equal 250.');
+    }
+
+    /****
+    * @description Validates start date on RD with empty current start date
+    */
+    @IsTest
+    private static void shouldSetStartDateToCurrentDateIfStartDateEmpty() {
+        Date startDate = Date.newInstance(2019, 11, 1);
+        npe03__Recurring_Donation__c rd = getRecurringDonationMonthlyBuilder()
+                .withStartDate(startDate)
+                .build();
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        insert rd;
+        Test.startTest();
+        rd.npe03__Amount__c = 250;
+        rd.StartDate__c = null;
+        update rd;
+        Test.stopTest();
+
         npe03__Recurring_Donation__c rd2 = rdGateway.getRecord(rd.Id);
-        System.assertEquals(Date.today(), rd2.StartDate__c, 'Start date should be today()');
+        System.assertEquals(Date.today(), rd2.StartDate__c, 'Next donation date should be today()');
```

Comment:
- No need to create a new var for the RD, the existing one can be reused. 
- The assert statement should be mentioning RD Effective Date.
```
rd = rdGateway.getRecord(rd.Id);
System.assertEquals(Date.today(), rd.StartDate__c, 'RD Effective Date should be set to today');
```